lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
+  else
+    return util._extend({}, this._headers);
```

Comment:
In this case, `getAllHeaders` should never be called in a hot loop, so optimizing should take a backseat to correctness – I don't want the user to be able to use the returned object to modify the outgoing headers, since that would be some spooky action at a distance. `util._extend` is ideal for this case.